origin-dapp/src/components/search/search-result.js

```diff
@@ -204,15 +204,19 @@ class SearchResult extends Component {
         }
       }
 
-      const searchResp = await origin.discovery.search(
-        this.props.query || '',
-        LISTINGS_PER_PAGE,
-        (this.state.page - 1) * LISTINGS_PER_PAGE,
-        Object.values(filters).flatMap(arrayOfFilters => arrayOfFilters)
-      )
+      const searchResp = await origin.discovery.search({
+        searchQuery: this.props.query || '',
+        numberOfItems: LISTINGS_PER_PAGE,
+        offset: (this.state.page - 1) * LISTINGS_PER_PAGE,
+        filters: Object.values(filters).flatMap(arrayOfFilters => arrayOfFilters)
+      })
 
       this.setState({
-        listingIds: searchResp.data.listings.nodes.map(listing => listing.id),
+        listings: searchResp.data.listings.nodes.map(listing => {
```

Comment:
This looks a bit hacky :)
Do we need to store the whole listing data in the state ? Or could could we just store both listingIds and featuredIds ?
e.g. something like this:
```
this.setState({
  listingIds: searchResp.data.listings.nodes.map(listing =>listing.id),
  featuredIds: searchResp.data.listings.nodes
    .filter(listing => listing.display === 'featured')
    .map(listing => listing.id)
  ...
})
```

Reply:
Yeah, since featured badge is in the recent change handled by the `listing-card` we don't need the `featuredIds` anymore. So going back to only storing `listingIds`. (what you propose in example minus the `featureIds` property)